tests/phpunit/tests/query.php

```diff
@@ -897,4 +897,64 @@ public function test_query_tag_404_does_not_throw_warning() {
 		$this->assertFalse( $q->is_tax() );
 		$this->assertFalse( $q->is_tag( 'non-existent-tag' ) );
 	}
+
+	/**
+	 * Test if $before_loop is true before loop.
+	 *
+	 * @ticket 58211
+	 */
+	public function test_before_loop_value_set_true_before_the_loop() {
+		// get a new query with 3 published posts.
+		$query = $this->get_new_wp_query_with_posts( 3 );
+
+		$this->assertTrue( $query->before_loop );
+	}
+
+	/**
+	 * Test $before_loop value is set to false when the loop starts.
+	 *
+	 * @ticket 58211
+	 */
+	public function test_before_loop_value_set_to_false_in_loop_with_post() {
+		// get a new query with 2 posts.
+		$query = $this->get_new_wp_query_with_posts( 2 );
+
+		while ( $query->have_posts() ) {
+			// $before_loop should be set false as soon as the_post is called for the first time.
+			$query->the_post();
+
+			$this->assertFalse( $query->before_loop );
+			break;
+		}
+	}
+
+	/**
+	 * Test $before_loop value is set to false when there is no post in the loop.
+	 *
+	 * @ticket 58211
+	 */
+	public function before_loop_set_false_after_loop_with_no_post() {
```

Comment:
Missing `test_` in function name.
```suggestion
	public function test_before_loop_set_false_after_loop_with_no_post() {
```

Reply:
Fixed.